internal/provision/provisionwatchers.go

```diff
@@ -48,43 +69,44 @@ func LoadProvisionWatchers(path string, dic *di.Container) errors.EdgeX {
 
 	var addProvisionWatchersReq []requests.AddProvisionWatcherRequest
 	for _, file := range files {
-		var watcher dtos.ProvisionWatcher
 		filename := filepath.Join(absPath, file.Name())
-		data, err := os.ReadFile(filename)
-		if err != nil {
-			lc.Errorf("Failed to read %s: %v", filename, err)
-			continue
-		}
+		processFile(lc, filename, nil, addProvisionWatchersReq)
+	}
 
-		if strings.HasSuffix(filename, yamlExt) || strings.HasSuffix(filename, ymlExt) {
-			err = yaml.Unmarshal(data, &watcher)
-			if err != nil {
-				lc.Errorf("Failed to YAML decode %s: %v", filename, err)
-				continue
-			}
-		} else if strings.HasSuffix(filename, jsonExt) {
-			err := json.Unmarshal(data, &watcher)
-			if err != nil {
-				lc.Errorf("Failed to JSON decode %s: %v", filename, err)
-				continue
-			}
-		} else {
-			continue
-		}
+	if len(addProvisionWatchersReq) == 0 {
+		return nil
+	}
 
-		err = common.Validate(watcher)
-		if err != nil {
-			lc.Errorf("ProvisionWatcher %s validation failed: %v", watcher.Name, err)
-			continue
-		}
+	pwc := container.ProvisionWatcherClientFrom(dic.Get)
+	ctx := context.WithValue(context.Background(), common.CorrelationHeader, uuid.NewString()) //nolint: staticcheck
+	_, edgexErr := pwc.Add(ctx, addProvisionWatchersReq)
+	return edgexErr
+}
 
-		if _, ok := cache.ProvisionWatchers().ForName(watcher.Name); ok {
-			lc.Infof("ProvisionWatcher %s exists, using the existing one", watcher.Name)
-		} else {
-			lc.Infof("ProvisionWatcher %s not found in Metadata, adding it...", watcher.Name)
-			req := requests.NewAddProvisionWatcherRequest(watcher)
-			addProvisionWatchersReq = append(addProvisionWatchersReq, req)
-		}
+func loadProvisionWatchersFromUri(path string, dic *di.Container) errors.EdgeX {
+	secretProvider := container.SecretProviderFrom(dic.Get)
+	bytes, err := file.Load(path, timeout, secretProvider)
+	if err != nil {
+		return errors.NewCommonEdgeX(errors.KindServerError, "failed to read directory", err)
+	}
+
+	if len(bytes) == 0 {
+		return nil
+	}
+
+	var files []string
+
+	err = json.Unmarshal(bytes, &files)
+	if err != nil {
+		return errors.NewCommonEdgeX(errors.KindServerError, "could not unmarshal provision watcher contents", err)
+	}
+
+	lc := container.LoggingClientFrom(dic.Get)
+	lc.Infof("Loading pre-defined provision watchers from %s(%d files found)", path, len(files))
+
+	var addProvisionWatchersReq []requests.AddProvisionWatcherRequest
+	for _, file := range files {
+		processFile(lc, file, secretProvider, addProvisionWatchersReq)
```

Comment:
Would be nice to know the name of the PW and check if it needs to be loaded before loading it from network when it might just be ignored. We could expand the index.json to include the PW name??
Something we should take to the Device Service WG for future enhancement.